mne/export/_edf.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+# Authors: MNE Developers
+#
+# License: BSD-3-Clause
+
+import numpy as np
+
+from ..utils import _check_edflib_installed, warn
+_check_edflib_installed()
+from EDFlib.edfwriter import EDFwriter  # noqa: E402
+
+
+def _try_to_set_value(header, key, value, channel_index=None):
+    """Helper function to set key/value pairs in EDF/BDF header."""
+    # all EDFLib set functions are set<X>
+    # for example "setPatientName()"
+    func_name = f'set{key}'
+    func = getattr(header, func_name)
+
+    # some setter functions are indexed by channels
+    if channel_index is None:
+        return_val = func(value)
+    else:
+        return_val = func(channel_index, value)
+
+    # a return value of 0 indicates an error
+    if return_val != 0:
+        raise RuntimeError(f"Setting {key} with {value} "
+                           f"returned an error")
+
+
+def _export_raw(fname, raw, physical_range, fmt):
+    """Export Raw objects to EDF or BDF files.
+
+    TODO:
+    - if Info stores transducer information, allow writing here.
+    - if Info stores techniciain information, allow writing here.
+    """
+    phys_dims = 'uV'
+
+    if fmt == 'bdf':
+        digital_min = -8388607
+        digital_max = 8388607
+    elif fmt == 'edf':
+        digital_min = -32767
+        digital_max = 32767
+
+    # load data first
+    raw.load_data()
+
+    # remove extra epoc and STI channels
+    drop_chs = ['epoc']
+    if not (raw.filenames[0].endswith('.fif')):
+        drop_chs.append('STI 014')
```

Comment:
what is that? at best I would look for channels of type stim and drop them.

Reply:
Done.